scr/main/java/bittercode/model/StoreException.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.IOException;
+
+import com.bittercode.constant.ResponseCode;
+
+public class StoreException extends IOException {
+
+    private String errorCode;
+    private String errorMessage;
+    private int statusCode;
+
+    public StoreException(String errorMessage) {
+        super(errorMessage);
+        this.errorCode = "BAD_REQUEST";
+        this.setStatusCode(400);
+        this.errorMessage = errorMessage;
+    }
+
+    public StoreException(ResponseCode errorCodes) {
+        super(errorCodes.getMessage());
+        this.statusCode = errorCodes.getCode();
+        this.errorMessage = errorCodes.getMessage();
+        this.setErrorCode(errorCodes.name());
+    }
+
+    public StoreException(String errroCode, String errorMessage) {
+        super(errorMessage);
+        this.errorCode = errroCode;
+        this.errorMessage = errorMessage;
+        this.statusCode = 422;
+    }
+
+    public StoreException(int statusCode, String errorCode, String errorMessage) {
+        super(errorMessage);
+        this.statusCode = statusCode;
+        this.errorCode = errorCode;
+        this.errorMessage = errorMessage;
+    }
+
+    public String getErrorCode() {
+        return null;
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/2#issuecomment-2419037001>#be98fd</a><span> - 10/17/2024, 09:42 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The getErrorCode() method is now returning null. This might lead to issues in error handling and debugging. Consider returning the actual error code or a default value if appropriate.
</div>
<div id="fix">
<b>Fix:</b> Modify the getErrorCode() method to return the actual error code. If the error code can be null in some cases, consider returning a default value or throwing an appropriate exception.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -41,3 +41,3 @@
  public String getErrorCode() {
 -        return null;
 +        return errorCode;
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged